### PR TITLE
feat: use localStorage when in dev mode

### PR DESCRIPTION
**Pull request by jwartofsky-yext**
When in devMode, LayoutEditor uses the theme from local storage.

When in devMode, ThemeEditor uses the layout from local storage.

Tested alongside changes to YextSitesStorm.
https://gerrit.yext.com/c/alpha/+/275033

Verified that when layout or theme is updated in dev mode, the change can be seen in the other mode.

### CODE REVIEW

File: src/internal/components/LayoutEditor.tsx [hunk 1]
```diff
@@ -63,7 +65,26 @@ export const LayoutEditor = (props: LayoutEditorProps) => {
    */
   useEffect(() => {
     devLogger.logData("THEME_DATA", themeData);
```
Comment (mkilpatrick) [inline src/internal/components/LayoutEditor.tsx:67]: Log the data actually being used, aka have a logger for the localStorage data and move this to the if themeData conditional.

File: src/utils/devLogger.ts [hunk 2]
```diff
@@ -3,7 +3,9 @@ export type DevLoggerPrefix =
   | "PUCK_CONFIG"
   | "SAVE_STATE"
   | "THEME_SAVE_STATE"
+  | "LOCAL_VISUAL_CONFIGURATION_DATA"
```
Comment (mkilpatrick) [inline src/utils/devLogger.ts:6]: We don't use this key when we use the local data in layout mode. We instead use VISUAL_CONFIGURATION_DATA with another log saying it's using localStorage. We should be consistent.

File: src/internal/components/ThemeEditor.tsx [hunk 3]
```diff
@@ -74,10 +74,7 @@ export const ThemeEditor = (props: ThemeEditorProps) => {
       const localHistories = JSON.parse(localHistoryArray) as History[];
       const localHistoryIndex = localHistories.length - 1;
       if (localHistoryIndex >= 0) {
-        devLogger.logData(
-          "LOCAL_VISUAL_CONFIGURATION_DATA",
-          localHistories[localHistoryIndex].state
-        );
+        devLogger.log("Theme Dev Mode - Using layout data from local storage");
```
Comment (mkilpatrick) [inline src/internal/components/ThemeEditor.tsx:77]: We still want to log the data though. Are we doing that?

Comment (jwartofsky-yext) [reply src/internal/components/ThemeEditor.tsx:77]: Whenever we receive themeData or visualConfigurationData it gets logged by the receiver.

We don't log in this specific case where we use the localStorage though, but there is a message to let the user know that it is using localStorage and they can access the message in console log if they wish.

**Review by mkilpatrick — APPROVED** [review]

**Review by mkilpatrick — APPROVED** [review]

**Pull request merged**